### PR TITLE
Sorting of search results by multiple ordered fields

### PR DESCRIPTION
**Pull request by bmichalski**
This pull request fixes the issue #365, in a slight different way than https://github.com/composer/packagist/pull/481.

It enables to sort by multiple ordered field, using HTML or JSON.

Even though the user interface does not permit it, as it may not be user friendly, the API does support a combination of ordering parameters.

Even if at the moment it may not be needed, it might come in handy when you decide to add other criterias to order by.

Below are some screenshots:

By downloads:

screenshot from 2015-05-04 21 11 47

By favorites:

screenshot from 2015-05-04 21 12 36

It also comes with automated tests.

Please be aware that these tests should not be run on a production machine, as they first drop / create the database to ensure isolation.

### CODE REVIEW

**Comment by stof** [comment]
This PR conflicts with master. Looks like you need a rebase

File: src/Packagist/WebBundle/Command/IndexPackagesCommand.php [hunk 1]
```diff
@@ -12,12 +12,16 @@
 
 namespace Packagist\WebBundle\Command;
 
+use DateTime;
+use Exception;
```
Comment (stof) [inline src/Packagist/WebBundle/Command/IndexPackagesCommand.php]: this goes against the Symfony coding standards, which are used in the Composer projects. It should be reverted

File: src/Packagist/WebBundle/Controller/WebController.php [hunk 2]
```diff
@@ -248,6 +248,111 @@ protected function setupPager($query, $page)
     }
 
     /**
+     * @param mixed $orderBys
```
Comment (stof) [inline src/Packagist/WebBundle/Controller/WebController.php]: this cannot be mixed. It requires passing an array

File: src/Packagist/WebBundle/Resources/public/js/search.js [hunk 3]
```diff
@@ -49,11 +57,57 @@
...
+
+            pushStateArguments = [
+                null,
+                'Search',
+                urlPrefix + '/search/?q=' + q + joinedOrderBysQryStrPart
+            ];
+
             if (firstQuery) {
-                window.history.pushState(null, "Search", "/search/?q=" + encodeURIComponent($('input[type="search"]', form).val()));
+                window.history.pushState.apply(window.history, pushStateArguments);
```
Comment (stof) [inline src/Packagist/WebBundle/Resources/public/js/search.js]: you don't really need to use ``.apply`` here. You could just use ``.pushState(null, "Search", url)``, after creating the ``url`` variable

**Comment by bmichalski** [comment]
I fixed the PR according to your comments.
I also merged composer/packagist master into my branch, there should no longer be a conflict.

**Comment by Seldaek** [comment]
Thanks @bmichalski! I did a couple tweaks as you can see in the commit linked above, but overall it's working quite well and will surely help people make better decisions.

**Pull request merged**